### PR TITLE
updated build system script commands

### PR DESCRIPTION
**Pull request by g4titanx**

### CODE REVIEW

**Comment by g4titanx** [comment]
do you have an idea of why the test is failing?

File: build_system/src/main.rs [hunk 1]
```diff
@@ -26,17 +26,23 @@ macro_rules! arg_error {
...
+Commands:
+
+        cargo     : Executes a cargo command. 
+        clean     : Cleans the build directory, removing all compiled files and artifacts.
+        prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
+        build     : Compiles the project. 
+        test      : Runs tests for the project.
+        info      : Displays information about the build environment and project configuration.
+        clone-gcc : Clones the GCC compiler from a specified source.
+        "
```
Comment (GuillaumeGomez) [inline build_system/src/main.rs:45]: The `rustc` command is missing.

File: build_system/src/main.rs [hunk 2]
```diff
@@ -34,11 +34,11 @@ Options:
         --help    : Displays this help message.
 
 Commands:
-
         cargo     : Executes a cargo command. 
+        rustc     : Compiles the program using the GCC compiler.
         clean     : Cleans the build directory, removing all compiled files and artifacts.
         prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
-        build     : Compiles the project. 
+        build     : Initiates the compilation process for the project, producing executable binaries.
```
Comment (GuillaumeGomez) [inline build_system/src/main.rs:41]: This is way too much information. Please put back the previous version.

File: build_system/src/main.rs [hunk 3]
```diff
@@ -26,17 +26,23 @@ macro_rules! arg_error {
...
+Commands:
+        cargo     : Executes a cargo command. 
+        rustc     : Compiles the program using the GCC compiler.
+        clean     : Cleans the build directory, removing all compiled files and artifacts.
+        prepare   : Prepares the environment for building, including fetching dependencies and setting up configurations.
+        build     : Compiles the project. 
+        test      : Runs tests for the project.
+        info      : Displays information about the build environment and project configuration.
+        clone-gcc : Clones the GCC compiler from a specified source.
+        "
```
Comment (GuillaumeGomez) [inline build_system/src/main.rs:45]: One last nit and it's all good for me, please change like this:

```suggestion
        clone-gcc : Clones the GCC compiler from a specified source."
```

Comment (g4titanx) [reply build_system/src/main.rs:45]: alright

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**